server/service/src/requisition/response_requisition/insert_program.rs

```diff
@@ -184,10 +208,70 @@ fn generate(
         .map(|line| line.item_id)
         .collect();
 
-    let requisition_line_rows =
+    let requisition_lines =
         generate_requisition_lines(ctx, &ctx.store_id, &requisition, program_item_ids)?;
 
-    Ok((requisition, requisition_line_rows))
+    let program_indicators = program_indicators(
+        connection,
+        Pagination::all(),
+        None,
+        Some(ProgramIndicatorFilter::new().program_id(EqualFilter::equal_to(&program.id))),
+    )?;
+
+    let customer_store = StoreRepository::new(connection)
+        .query_one(StoreFilter::new().name_id(EqualFilter::equal_to(&other_party_id)))?;
+
+    let generate_indicator_values = generate_program_indicator_values(
+        &ctx.store_id,
+        &period_id,
+        &other_party_id,
+        program_indicators,
+    );
+
+    let indicator_values = match customer_store {
+        Some(_) => generate_indicator_values,
+        None => vec![],
+    };
+
+    Ok(GenerateResult {
+        requisition,
+        requisition_lines,
+        indicator_values,
+    })
+}
+
+fn generate_program_indicator_values(
+    store_id: &str,
+    period_id: &str,
+    other_party_id: &str,
+    program_indicator: HashMap<String, ProgramIndicator>,
```

Comment:
Tiniest nit: isn't this plural `program_indicators`?

```suggestion
    program_indicators: HashMap<String, ProgramIndicator>,
```